# HG changeset patch
# User Torsten Jager <t.jager@gmx.de>
# Date 1571675945 -7200
#      Mon Oct 21 18:39:05 2019 +0200
# Node ID ba74937f17e6ff23fb77a7e7f895d9fa09aae8dc
# Parent  5a9389300b8edc08e8732bf044bd70d6190fe024
Fix build with old libxine.

diff -r 5a9389300b8e -r ba74937f17e6 configure.ac
--- a/configure.ac	Sun Aug 18 13:15:44 2019 +0300
+++ b/configure.ac	Mon Oct 21 18:39:05 2019 +0200
@@ -91,6 +91,7 @@
 XINE_LIB_SHIMS
 XINE_LIB_OPEN_CLOEXEC
 XINE_LIB_SOCKET_CLOEXEC
+XINE_LIB_LIST_NEXT_VALUE
 
 CFLAGS="${CFLAGS} ${XINE_CFLAGS}"
 
diff -r 5a9389300b8e -r ba74937f17e6 m4/_xine.m4
--- a/m4/_xine.m4	Sun Aug 18 13:15:44 2019 +0300
+++ b/m4/_xine.m4	Mon Oct 21 18:39:05 2019 +0200
@@ -241,3 +241,21 @@
     LIBS="$tmp_LIBS"
     ])
 
+dnl Test for xine_list_next_value function
+AC_DEFUN([XINE_LIB_LIST_NEXT_VALUE],
+    [AC_MSG_CHECKING([for xine_list_next_value within xine-lib])
+    tmp_CFLAGS="$CFLAGS"
+    tmp_LIBS="$LIBS"
+     CFLAGS="$CFLAGS $XINE_CFLAGS"
+     LIBS="$LIBS $XINE_LIBS"
+     AC_LINK_IFELSE(
+         [AC_LANG_PROGRAM([
+         ],[
+xine_list_next_value ();
+         ])],
+         [AC_DEFINE([HAVE_XINE_LIST_NEXT_VALUE], [1], [Define if xine-lib supports xine_list_next_value])
+          AC_MSG_RESULT([yes])],
+         [AC_MSG_RESULT([no])])
+    CFLAGS="$tmp_CFLAGS"
+    LIBS="$tmp_LIBS"
+   ])
diff -r 5a9389300b8e -r ba74937f17e6 src/xitk/setup.c
--- a/src/xitk/setup.c	Sun Aug 18 13:15:44 2019 +0300
+++ b/src/xitk/setup.c	Mon Oct 21 18:39:05 2019 +0200
@@ -31,6 +31,17 @@
 #include "common.h"
 
 #include <xine/list.h>
+#ifdef HAVE_XINE_LIST_NEXT_VALUE
+#  define _xine_list_next_value(_xlnv_list,_xlnv_ite) xine_list_next_value (_xlnv_list, _xlnv_ite)
+#else
+static inline void *_xine_list_next_value (xine_list_t *list, xine_list_iterator_t *ite) {
+  if (*ite)
+    *ite = xine_list_next (list, *ite);
+  else
+    *ite = xine_list_front (list);
+  return *ite ? xine_list_get_value (list, *ite) : NULL;
+}
+#endif
 
 
 #define WINDOW_WIDTH             630
@@ -809,7 +820,7 @@
     xitk_widget_t *sw;
     xine_list_iterator_t ite = NULL;
     while (1) {
-      sw = xine_list_next_value (setup->widgets, &ite);
+      sw = _xine_list_next_value (setup->widgets, &ite);
       if (!ite)
         break;
       xitk_destroy_widget (sw);
